homework5.py

```diff
@@ -0,0 +1,102 @@
+#и название дня недели (например, 139.18.150.126 23:12:44 sunday). Создайте новый текстовый файл, который 
+#бы содержал список ip без повторений из первого файла. Для каждого ip укажите количество посещений, 
+#наиболее популярный день недели. Последней строкой в файле добавьте наиболее популярный отрезок времени 
+#в сутках длиной один час в целом для сайта.
+import csv
+from datetime import *
+
+#----------------------------------------------------------------------------------#
+# 1
+#----------------------------------------------------------------------------------#
+def task1():
+    new_lines = []
+    with open(r'.\homework5_data\1.txt', 'r', encoding='utf-8') as f:
+        for line in f:
+            unwanted_words = [unwanted_word for unwanted_word in line.split() if 3 <= len(unwanted_word.strip('.,?!()\'')) <= 5]
+            unwanted_word_count = len(unwanted_words)
+            unwanted_word_count = unwanted_word_count if unwanted_word_count % 2 == 0 else unwanted_word_count - 1
+            clean_line = line
+            for i in range(unwanted_word_count):
+                clean_line = clean_line.replace(unwanted_words[i] + ' ', '', 1)
+            new_lines.append(clean_line)
+
+    with open(r'.\homework5_data\1.txt', 'w', encoding='utf-8') as f:
+        f.writelines(new_lines)
+
+
+#----------------------------------------------------------------------------------#
+# 2
+#----------------------------------------------------------------------------------#
+def task2():
+    file_path = r'.\homework5_data\2_input.txt'
+    required_phones = []
+
+    with open(r'.\homework5_data\2_input.txt', 'r', encoding='utf-8') as f:
+        required_phones = [line.split()[2] + '\n' for line in f if line.split()[1][0] in ['С', 'К']]
+
+    with open(r'.\homework5_data\2_output.txt', 'w', encoding='utf-8') as f:
+        f.writelines(required_phones)
+
+
+def task3():
+    #----------------------------------------------------------------------------------#
+    # 3
+    #----------------------------------------------------------------------------------#
+    file_path = r'.\homework5_data\3.txt'
+    new_lines = []
+
+    with open(file_path, 'r', encoding='utf-8') as f:
+        current_lines = list(f)
+        max_len = len(max(current_lines, key=len))
+        new_lines = [string.rjust(max_len, ' ') for string in current_lines]
+
+    with open(file_path, 'w', encoding='utf-8') as f:
+        f.writelines(new_lines)
+
+
+#----------------------------------------------------------------------------------#
+# 4
+#----------------------------------------------------------------------------------#
+def task4():
+    input_file_path = r'.\homework5_data\4_input.csv'
+    output_file_path = r'.\homework5_data\4_output.txt'
+    output_data = []
+    
+    with open(input_file_path, newline='', encoding='utf-8') as csvfive:
+        reader = csv.DictReader(csvfive)
```

Comment:
if you would like to be prepared to read csv files, then, probably would be a good idea to check the format of this file before this line